# Data Exploration on Twitter Suicidal Data

Source: https://www.kaggle.com/datasets/hosammhmdali/twitter-suicidal-data

In [1]:
import pandas as pd

In [2]:
FILEPATH = "twitter-suicidal_data.csv"

In [3]:
# Load the dataset from the CSV file
df = pd.read_csv(FILEPATH)

# Display the first few rows of the dataframe
df.head()

tweet  intention
0  my life is meaningless i just want to end my l...          1
1  muttering i wanna die to myself daily for a fe...          1
2  work slave i really feel like my only purpose ...          1
3  i did something on the 2 of october i overdose...          1
4  i feel like no one cares i just want to die ma...          1

In [4]:
print(f"Length of Dataset: {len(df)}")
print(f"Columns: {df.columns}")

Length of Dataset: 9119
Columns: Index(['tweet', 'intention'], dtype='object')


In [5]:
df.shape

(9119, 2)

There are 9119 records (i.e., posts) in this dataset, and each one is represented by 2 attributes.

In [6]:
print(f"Unique Class: {df['intention'].unique()}")

Unique Class: [1 0]


In [7]:
print(f"Number of Records of Class 0: {len(df[df['intention'] == 0])}")
print(f"Number of Records of Class 1: {len(df[df['intention'] == 1])}")

Number of Records of Class 0: 5121
Number of Records of Class 1: 3998


## Meaning of Each Class

In [8]:
# Extract 10 records for each class
df_class_0 = df[df['intention'] == 0].head(10)
df_class_1 = df[df['intention'] == 1].head(10)

In [9]:
df_class_0

tweet  intention
3325   i wish i got to watch it with you i miss you ...          0
3326   i want to go to promote gear and groove but u...          0
3327  oh manwas ironing fave top to wear to a meetin...          0
3328  sadly though i ve never gotten to experience t...          0
3329  wonders why someone that u like so much can ma...          0
3330                    strider is a sick little puppy           0
3331  i m sooo sad they killed off kutner on house w...          0
3332  this week just seems to get longer and longer ...          0
3333  was intending to finish editing my 536page nov...          0
3334  oh just got all my macheist 30 apps sweet didn...          0

In [10]:
df_class_1

tweet  intention
0  my life is meaningless i just want to end my l...          1
1  muttering i wanna die to myself daily for a fe...          1
2  work slave i really feel like my only purpose ...          1
3  i did something on the 2 of october i overdose...          1
4  i feel like no one cares i just want to die ma...          1
5  i am great and wonderful i am worth it except ...          1
6  i ll be dead just you wait and see my last wor...          1
7  health anxiety prompting some bad thoughts in ...          1
8  everything is okay but nothing feels okay i ve...          1
9  ptsd and alcohol i had some extremely horrible...          1

Labels:
* 0 = non-suicidal 
* 1 = suicidal